app/models/user.rb

```diff
@@ -2023,8 +2023,11 @@ def self.purge_unactivated
     destroyer = UserDestroyer.new(Discourse.system_user)
 
     User
+      .joins(
+        "LEFT JOIN user_histories ON user_histories.target_user_id = users.id AND action = #{UserHistory.actions[:deactivate_user]} AND acting_user_id != #{Discourse.system_user.id}",
```

Comment:
Slight tweak to this, maybe say `acting_user_id > 0`, which indicates any non-bot user.